### PR TITLE
iOS-5510: Fix warnings in LHPasscodeViewController.

### PR DESCRIPTION
**Pull request by lancy-mega**
Fixing warnings in LHPasscodeViewController.

### CODE REVIEW

File: LTHPasscodeViewController/LTHPasscodeViewController.m [hunk 1]
```diff
@@ -39,7 +39,8 @@ @implementation UIApplication (window)
 + (UIWindow *)currentWindow {
     UIWindow *window;
 #ifdef LTH_IS_APP_EXTENSION
-    window = [UIApplication sharedApplication].keyWindow;
+    NSPredicate *isKeyWindow = [NSPredicate predicateWithFormat:@"isKeyWindow == YES"];
+    window = [UIApplication.sharedApplication.windows filteredArrayUsingPredicate:isKeyWindow].firstObject;
```
Comment (simonwng) [inline LTHPasscodeViewController/LTHPasscodeViewController.m]: `@property(class, nonatomic, readonly) UIApplication *sharedApplication NS_EXTENSION_UNAVAILABLE_IOS("Use view controller based solutions where appropriate instead.");
`
sharedApplication is not available on extensions?

Comment (lancy-mega) [reply LTHPasscodeViewController/LTHPasscodeViewController.m]: Thanks! I will have updated the code. Please have a look and let me know your thoughts. Thanks!

File: LTHPasscodeViewController/LTHPasscodeViewController.h [hunk 2]
```diff
@@ -352,12 +352,17 @@ typedef NS_ENUM(NSInteger, PasscodeType) {
 
 // MARK: - Methods
 
+#ifndef LTH_IS_APP_EXTENSION
```
Comment (jnavarrom) [inline LTHPasscodeViewController/LTHPasscodeViewController.h:355]: Does passcode work in share and picker extension after applying this change?

Comment (lancy-mega) [reply LTHPasscodeViewController/LTHPasscodeViewController.h:355]: | Before  | After |
| ------------- | ------------- |
|   |   |
| <video src="https://user-images.githubusercontent.com/53881615/141007180-f53e6bb4-65fc-4fa8-b827-474e71035ad0.MP4" width=320>  | <video src="https://user-images.githubusercontent.com/53881615/141007154-50bf6c25-eb7e-4899-9c7c-788be04ac28b.MP4" width=320>  |

File: LTHPasscodeViewController/LTHPasscodeViewController.m [hunk 3]
```diff
@@ -37,18 +39,15 @@ + (nullable UIWindow *)currentWindow;
...
-    return window;
+    return [UIApplication sharedApplication].windows.firstObject;
 }
 
 @end
 
+#endif
+
 @interface LTHPasscodeViewController () <UITextFieldDelegate>
+@property (nonatomic, weak)   UIView      *presentingView;
```
Comment (simonwng) [inline LTHPasscodeViewController/LTHPasscodeViewController.m:51]: Do we need to make this `presentingView` only available for `LTH_IS_APP_EXTENSION`? so for non-extensions, we always use key window

File: LTHPasscodeViewController/LTHPasscodeViewController.m (hunk 1 above)
Comment (simonwng) [reply LTHPasscodeViewController/LTHPasscodeViewController.m]: Nice, thanks

**Review by simonwng — APPROVED** [review]

File: LTHPasscodeViewController/LTHPasscodeViewController.h (hunk 2 above)
Comment (jnavarrom) [reply LTHPasscodeViewController/LTHPasscodeViewController.h:355]: Thanks for checking Lancy

**Pull request merged**